src/util/arr.h

```diff
@@ -266,6 +266,17 @@ static void array_free(array_t arr) {
     arr[--(array_hdr(arr)->len)];    \
   })
 
+/* Remove a specified element from the array */
+#define array_del(arr, ix)                                        \
```

Comment:
You can just replace this element with the last element, it will be more efficient, if you do it make sure to document it so we will know (when using it) that it does not promise to keep the order.

Reply:
an array *does* promise to keep the order, unfortunately. I suppose I could make another function that does this, though. e.g. `array_swap_del`